src/Microsoft.Health.Dicom.Api/Modules/SecurityModule.cs

```diff
@@ -71,24 +72,25 @@ public void Load(IServiceCollection services)
 
                 if (_securityConfiguration.Authorization.Enabled)
                 {
-                    services.Add<RoleLoader>().Transient().AsImplementedInterfaces();
+                    services.Add<DicomRoleLoader>().Transient().AsImplementedInterfaces();
                     services.AddSingleton(_securityConfiguration.Authorization);
 
-                    services.AddSingleton<IDicomAuthorizationService, RoleBasedDicomAuthorizationService>();
+                    services.AddSingleton<IAuthorizationService<DataActions>, RoleBasedAuthorizationService<DataActions, IDicomRequestContext>>();
                 }
                 else
                 {
-                    services.AddSingleton<IDicomAuthorizationService, DisabledDicomAuthorizationService>();
+                    services.AddSingleton<IAuthorizationService<DataActions>, DisabledAuthorizationService<DataActions>>();
                 }
             }
             else
             {
-                services.AddSingleton<IDicomAuthorizationService, DisabledDicomAuthorizationService>();
+                services.AddSingleton<IAuthorizationService<DataActions>, DisabledAuthorizationService<DataActions>>();
```

Comment:
Ah, the existing versions of the shared packages have these types?

Reply:
Yep, these were abstracted away.